### ISSUE TITLE
TypeError during Parallel Execution 

### ISSUE THREAD

**Issue by chirag-fynd** [reportportal/agent-js-playwright#99]
We have observed below error while doing parallel execution of tests on multiple workers. Anything we are missing ?
``` typescript
/Users/xyz/qa-automation/projectname/node_modules/@reportportal/client-javascript/lib/report-portal-client.js:516
              this.map[tempId].realId = response.id;
                                      ^

TypeError: Cannot set properties of undefined (setting 'realId')
    at /Users/xyz/qa-automation/projectname/node_modules/@reportportal/client-javascript/lib/report-portal-client.js:516:39
    at processTicksAndRejections (node:internal/process/task_queues:95:5)

```

**Comment by AmsterGet** [comment]
Hello @chirag-fynd !
Seems like it is an issue with access to an already finished test item.
Could you provide a minimal reproducible sample to help us debug this issue?

**Comment by chirag-fynd** [comment]
@AmsterGet 

Below is Report Configuration in playwright 

```typescript
const RPconfig = {
    token: goldBugConfig.rpToken,
    endpoint: goldBugConfig.rpEndpoint,
    project:  goldBugConfig.rpProject,
    launch: 'Playwright Automation',
    includeTestSteps: true,
     attributes: [
       {
         key: 'ENVIRONMENT: ',
         value: goldBugConfig.platformUrl,
       },
       {
        key: 'TAGS: ',
        value: goldBugConfig.testCaseTag,
      },
      {
        key: 'Triggered by: ',
        value: process.env.GITLAB_USER_NAME ? process.env.GITLAB_USER_NAME : 'test user',
      },
      {
        key: 'Triggered Type: ',
        value: process.env.CI_PIPELINE_SOURCE ? process.env.CI_PIPELINE_SOURCE : 'testing' ,
      },
    ],
    description: 'Pipeline URL : '  + process.env.CI_PIPELINE_URL,
  };
console.log(RPconfig,"RP config")

 reporter: [
    ['list'],
    ['html', { open: 'never' }],
    ['@reportportal/agent-js-playwright', RPconfig]
  ],
```

we have below configuration for parallel runs for different suites.

```typescript
  fullyParallel: true,
  retries: process.env.CI ? 1 : 0,
  workers: process.env.CI ? 3 : 4,
```

Test Specs are written in describe blocks for different test cases in test suites for serial mode runs

```typescript
test.describe.configure({ mode: 'serial' });

test.describe('CBS', async () => {
    test('Test case 1 ', async () => {
     // Test case steps
    });
`});`
```
    

Let me know if anything else is required. Thanks.

**Comment by chirag-fynd** [comment]
Please find below screenshot for the error.

<img width="931" alt="Screenshot 2023-05-23 at 11 57 44 AM" src="https://github.com/reportportal/agent-js-playwright/assets/107979613/228401e3-2dda-4431-b15b-ce6fb38f3b8f">

**Comment by Pruthvisk** [comment]
Hi team do we have any work around for this

**Comment by AmsterGet** [comment]
Hello guys @chirag-fynd @Pruthvisk !
I tried to reproduce your issue on my own, but failed.
If you can create a reproducible sample it will definitely help us solve this problem.
Alternatively, you can fork our [playwright example] (https://github.com/reportportal/examples-js/tree/master/example-playwright) and update its tests to provide us with a reproducible example.

**Comment by AmsterGet** [comment]
Hello @chirag-fynd, @Pruthvisk !
Did you have a chance to create an issue example?

**Comment by PodarDelia** [comment]
I have this issue too. I am using nested steps and the error is: 

Failed to finish nested step. Error: Request failed with status code 503
    at /src/node_modules/@reportportal/client-javascript/lib/rest.js:41:15
    at processTicksAndRejections (node:internal/process/task_queues:95:5)
/src/node_modules/@reportportal/client-javascript/lib/report-portal-client.js:512
              this.map[tempId].realId = response.id;
                                      ^
TypeError: Cannot set properties of undefined (setting 'realId')

**Comment by AmsterGet** [comment]
Hello @PodarDelia !
Could you check the availability of the ReportPortal API service?
It looks like in your case the issue was due to the API service being unavailable, according to the response status code.

**Comment by AmsterGet** [comment]
@chirag-fynd @Pruthvisk 
Have you also observed response errors or timeout errors?

**Comment by PodarDelia** [comment]
Hello @AmsterGet ,
I checked and it looks ok. I have to mention that just sometimes i get this error.

**Comment by AmsterGet** [comment]
Hello guys!
@chirag-fynd , @Pruthvisk , @PodarDelia 
Since we still haven't been able to reproduce it on our side, it would be very helpful to get a reproducible sample from your side to speed up the fix :)

**Comment by AmsterGet** [comment]
Hi guys!
The issue has been fixed as part of [client-javascript version 5.0.14](https://github.com/reportportal/client-javascript/releases/tag/v5.0.14). It is already available in the latest version of the agent, so please try it.
I'm closing this issue, if you have any other questions, feel free to open a new one.
